### DISCUSSION TITLE
Split pdf according to results bytes sizes

### DISCUSSION THREAD

**Discussion by PasaOpasen** [pymupdf/PyMuPDF#2205] [Q&A] [answered]
What is the fastest way to split pdf to several documents with size < X?

Is there a way to get approx bytes size of each document page? I know that there can be repeating images or fonts, so maybe there is a way for accumulative size getting? For example:

1st page = 50 bytes
1st + 2nd = 80 bytes
1 + 2 + 3 = 110 bytes > 100 -> move 1+2 to separate document

3rd page = 40 bytes
3 + 4 = 70 bytes
...

**Comment by JorjMcKie** [comment, marked as answer]
This is unpredictable. There are even more cases than you would believe, where splitting a PDF into its `n` single pages delivers `n` one-page PDFs which **_have each the same size as the original_**!

The PDF sub-document is created (and must be so created) by including all xrefs that are referenced by some page of the chosen subset. So if evry page refers to some central object, which in turn refers to other pages, then those other pages (and their referrals) will become part of the current one-page subset, too.
Crazy - but it happens!

**Reply by PasaOpasen** [reply]
@JorjMcKie So, I will get a document size only after saving?

**Reply by JorjMcKie** [reply]
Correct!
So maybe a suitable approach is to first make 1-page sub-documents, create a `.tobytes()` memory file, and while below your threshold size, join it with the previous one.
**Or** just divide the current size by your threshold size and use the result as a good-enough approximation.
